pkg/processor/runtime/rpc/connection/abstract.go

```diff
@@ -327,52 +385,80 @@ func (be *AbstractEventConnection) processItem(item interface{}, functionLogger
 		be.functionLogger = nil
 		return nil, errors.Wrapf(err, "Can't encode item: %+v", item)
 	}
-	processingResults, ok := <-be.resultChan
 
+	// if eventTimeout is 0, we wait for response without timeout
+	// if it is set, we wait either for response or the timeout
+	if be.connectionManager.GetConfig().eventTimeout == 0 {
+		processingResults, ok := <-be.resultChan
+		return be.postProcessEventRegularFlow(processingResults, ok)
+	} else {
+		return be.waitForResponseWithTimeout()
+	}
+}
+
+func (be *AbstractEventConnection) waitForResponseWithTimeout() (*result.BatchedResults, error) {
+	ticker := time.NewTicker(be.connectionManager.GetConfig().eventTimeout)
+	defer ticker.Stop()
+
+	select {
+	case processingResults, isClientDisconnected := <-be.resultChan:
+		return be.postProcessEventRegularFlow(processingResults, isClientDisconnected)
+	case <-ticker.C:
+		return be.postProcessEventOnTimeout()
+	}
```

Comment:
What happens to the result coming from the `be.resultChan` in this case?
There will be no one reading from it, is this handled in the restart flow (in `Replace`)?

Reply:
@TomerShor yes, during replace this connection is closed (and channel as well)